### PR TITLE
Add job.state endpoint

### PR DESCRIPTION
**Pull request by dahal**
Add `job.state` endpoint which will give :point_down: response:-

``` ruby
{
  "id":000000,
  "options":{
  "logical_aggregation":true,
  "track_clones":true,
  "include_unfinished":true,
  "front_load":false,
  "after_gold":10
  },
  "title":null,
  "secret":"redacted",
  "project_number":null,
  "alias":null,
  "judgments_per_unit":3,
  "units_per_assignment":5,
  "pages_per_assignment":1,
  "max_judgments_per_worker":null,
  "max_judgments_per_ip":null,
  "gold_per_assignment":0,
  "minimum_account_age_seconds":null,
  "execution_mode":"worker_ui_remix",
  "payment_cents":10,
  "custom_key":null,
  "design_verified":true,
  "require_worker_login":null,
  "public_data":false,
  "variable_judgments_mode":"none",
  "max_judgments_per_unit":null,
  "expected_judgments_per_unit":null,
  "min_unit_confidence":null,
  "units_remain_finalized":null,
  "auto_order_timeout":null,
  "auto_order_threshold":null,
  "completed_at":null,
  "state":"unordered",
  "auto_order":false,
  "webhook_uri":null,
  "send_judgments_webhook":null,
  "language":"en",
  "minimum_requirements":null,
  "desired_requirements":null,
  "order_approved":true,
  "max_work_per_network":null,
  "copied_from":null,
  "created_at":"2014-03-21T03:10:54+00:00",
  "updated_at":"2014-03-21T03:10:54+00:00",
  "included_countries":[
  ],
  "excluded_countries":[
  ],
  "instructions":"",
  "cml":null,
  "js":null,
  "css":null,
  "problem":null,
  "confidence_fields":null,
  "gold":{
  },
  "units_count":0,
  "golds_count":0,
  "judgments_count":0,
  "support_email":"jdoe@crowdflower.com",
  "crowd_costs":0.0,
  "completed":false,
  "fields":null
}
```

```
```

### CODE REVIEW

File: lib/crowdflower/job.rb [hunk 1]
```diff
@@ -74,6 +74,10 @@ def status
       connection.get("#{resource_uri}/#{@id}/ping")
     end
     
+    def state
+      connection.get("#{resource_uri}/#{@id}")
+    end
+
```
Comment (cf-hound) [inline lib/crowdflower/job.rb:80]: Trailing whitespace detected.

File: lib/crowdflower/job.rb [hunk 2]
```diff
@@ -132,19 +136,19 @@ def disable_channel(channel_name)
...
     end
 
     def add_tags(tags)
-      connection.post("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } ) 
+      connection.post("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } )
     end
 
     def remove_tags(tags)
-      connection.delete("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } ) 
+      connection.delete("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } )
```
Comment (cf-hound) [inline lib/crowdflower/job.rb:151]: Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside parentheses detected.

File: lib/crowdflower/job.rb [hunk 3]
```diff
@@ -132,19 +136,19 @@ def disable_channel(channel_name)
...
     end
 
     def update_tags(tags)
-      connection.put("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } ) 
+      connection.put("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } )
     end
 
     def add_tags(tags)
-      connection.post("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } ) 
+      connection.post("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } )
```
Comment (cf-hound) [inline lib/crowdflower/job.rb:147]: Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside parentheses detected.

File: lib/crowdflower/job.rb [hunk 4]
```diff
@@ -132,19 +136,19 @@ def disable_channel(channel_name)
     end
 
     def tags
-      connection.get("#{resource_uri}/#{@id}/tags") 
+      connection.get("#{resource_uri}/#{@id}/tags")
     end
 
     def update_tags(tags)
-      connection.put("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } ) 
+      connection.put("#{resource_uri}/#{@id}/tags", {:body => { :tags => tags } } )
```
Comment (cf-hound) [inline lib/crowdflower/job.rb:143]: Redundant curly braces around a hash parameter.<br>Use the new Ruby 1.9 hash syntax.<br>Space inside { missing.<br>Space inside parentheses detected.

File: lib/crowdflower/job.rb [hunk 5]
```diff
@@ -28,8 +28,8 @@ def self.upload(file, content_type, job = nil, opts = {})
       job_uri = job ? "/#{job.kind_of?(Job) ? job.id : job}" : ""
       conn = job.kind_of?(Job) ? job.connection : self.connection
       upload_uri = "#{resource_uri}/#{job_uri}/upload".squeeze("/")
-      res = conn.post(upload_uri, 
-        :body => File.read(file), 
+      res = conn.post(upload_uri,
+        :body => File.read(file),
```
Comment (cf-hound) [inline lib/crowdflower/job.rb:32]: Align the parameters of a method call if they span more than one line.<br>Use the new Ruby 1.9 hash syntax.

File: lib/crowdflower/job.rb [hunk 6]
```diff
@@ -74,6 +74,10 @@ def status
       connection.get("#{resource_uri}/#{@id}/ping")
     end
     
+    def state
```
Comment (arvel) [inline lib/crowdflower/job.rb:77]: This new method is redundant with the pre-existing get method Line #48

Comment (mrgordon) [inline lib/crowdflower/job.rb:77]: Good point

**Comment by mrgordon** [comment]
As @arvel points out, this is the basic job `get`, not a `state` specific endpoint. That is why `state` comes back as part of the payload along with the rest of the keys. You can just use the existing method.

**Comment by dahal** [comment]
It makes sense, but it would be nice to add `job.get` 's response on docs (readme). Instead of listing each and everything [like this](https://github.com/CrowdFlower/ruby-crowdflower#get-example-jobs-json-which-the-get-method-has-access-to-httpscrowdflowercomjobs418404json)

And its upto user to determine what s/he wants to use.

**Comment by mrgordon** [comment]
Point taken. That part seems reasonable although the keys change over time so I see several in your example that wouldn't be there for a new job as we no longer support them. I wouldn't discourage adding an example response with a note that it may be somewhat out of date if you find it useful though

**Pull request closed without merging**